### PR TITLE
RSDK-8247 Implement gpio output as PWM in generic linux

### PR DESCRIPTION
**Pull request by oliviamiller**
Setting 0 or 100% duty cycle as gpio output on pwm pins. Adds an error if calling getGPIO on one of these pins. Tested on the pi5 and beaglebone, works great!

### CODE REVIEW

File: components/board/beaglebone/data.go [hunk 1]
```diff
@@ -71,10 +71,10 @@ var boardInfoMappings = map[string]genericlinux.BoardInformation{
 			{Name: "911", DeviceName: "gpiochip1", LineNumber: 1, PwmChipSysfsDir: "", PwmID: -1},
 			{Name: "912", DeviceName: "gpiochip1", LineNumber: 45, PwmChipSysfsDir: "", PwmID: -1},
 			{Name: "913", DeviceName: "gpiochip1", LineNumber: 2, PwmChipSysfsDir: "", PwmID: -1},
-			// We're unable to get GPIO to work on pin 914 despite Beaglebone's docs. PWM works.
+			// Note that gpio input cannot be used on this pin.
 			{Name: "914", DeviceName: "gpiochip1", LineNumber: 93, PwmChipSysfsDir: "3020000.pwm", PwmID: 0},
 			{Name: "915", DeviceName: "gpiochip1", LineNumber: 47, PwmChipSysfsDir: "", PwmID: -1},
-			// We're unable to get GPIO to work on pin 916 despite Beaglebone's docs. PWM works.
+			// Note that gpio input cannot be used on this pin.
```
Comment (penguinland) [inline components/board/beaglebone/data.go:77]: :+1: Thanks for updating all these comments!

File: components/board/genericlinux/gpio.go [hunk 2]
```diff
@@ -122,6 +123,19 @@ func (pin *gpioPin) Set(ctx context.Context, isHigh bool,
 // This function assumes you've already locked the mutex. It sets the value of a pin without
 // changing whether the pin is part of a software PWM loop.
 func (pin *gpioPin) setInternal(isHigh bool) (err error) {
+	// if pin is a hw pwm pin, set to either 0% or 100% duty cycle
+	if pin.hwPwm != nil {
+		var value float64
+		if isHigh {
+			value = 1
```
Comment (penguinland) [inline components/board/genericlinux/gpio.go:130]: Mild preference to use `1.0` to emphasize that this is a float and not an int.

File: components/board/genericlinux/gpio.go [hunk 3]
```diff
@@ -122,6 +123,19 @@ func (pin *gpioPin) Set(ctx context.Context, isHigh bool,
...
+		}
+		err := pin.hwPwm.SetPwm(10, value)
```
Comment (penguinland) [inline components/board/genericlinux/gpio.go:132]: What's the 10 on this line? I think it's the frequency, and then I wonder why we're changing that from whatever the old frequency was. I suspect the answer is "if the old frequency is unset, we need to make it some legitimate value, and 10 is good enough. If the old frequency was already set, it's okay to change it to some other value here, and we'll change it back if someone sets it again later. Note that we're not modifying `pin.pwmFreqHz`, so the old value is still available for later use." I'd put in a comment like that in the code, but mine is too wordy.

**Review by penguinland — APPROVED** [review]
LGTM! Nice work turning this out so quickly.

File: components/board/genericlinux/gpio.go (hunk 3 above)
Comment (oliviamiller) [reply components/board/genericlinux/gpio.go:132]: will add a comment about this!

Comment (mariapatni) [reply components/board/genericlinux/gpio.go:136]: Is there any specific reason why the frequency is set to 10? I guess I'm wondering if it should be lower/higher for performance reasons / to properly recreate GPIO. I'm assuming the difference in frequency is negligible but wanted to ask anyways!

**Review by mariapatni — APPROVED** [review]
LGTM!
